docs/dependencies.md

```diff
@@ -0,0 +1,287 @@
+  production environment is provisioned/installed, it should end up
+  using the exactly correct versions of all major dependencies.
+* **Low maintenance burden**.  To the extent possible, we want to
+  avoid manual work and keeping track of things that could be
+  automated.  This makes it easy to keep running the latest versions
+  of our various dependencies.
+
+The purpose of this document is to detail all of Zulip's third-party
+dependencies and how we manage their versions.
+
+## Philosophy on adding third-party dependencies
+
+In the Zulip project, we take a pragmatic approach to third-party
+dependencies.  Overall, if a third-party project does something well
+that Zulip needs to do (and has an appropriate license), we'd love to
+use it rather than reinventing the wheel.  If the third-party project
+needs some small changes to work, we prefer to make those changes and
+contribute them upstream.  When the upstream maintainer is slow to
+respond, we may use a fork of the dependency until the code is merged
+upstream; as a result, we usually have a few packages in
+`requirements.txt` that are installed from a GitHub URL.
+
+What we look for in choosing dependencies is whether the project is
+well-maintained.  Usually one can tell fairly quickly from looking at
+a project's issue tracker how well-managed it is: a quick look at how
+the issue tracker is managed (or not) and the test suite is usually
+enough to decide if a project is going to be a high-maintenance
+dependency or not.  That said, we do still take on some smaller
+dependencies that don't have a well-managed project, if we feel that
+using the project will still be a better investment than writing our
+own implementation of that project's functionality.  We've adopted a
+few projects in the past that had a good codebase but whose maintainer
+no longer had time for them.
+
+One case where we apply added scrutiny to third-party dependencies is
+JS libraries.  They are a particularly important concern because we
+want to keep the Zulip webapp's JS bundle small, so that Zulip
+continues to load quickly on systems with low network bandwidth.
+We'll look at large JS libraries with much greater scrutiny for
+whether their functionality justifies their size than Python
+dependencies, since an extra 50KB of code usually doesn't matter in
+the backend, but does in JavaScript.
+
+## System packages
+
+For the third-party services like Postgres, Redis, Nginx, and RabbitMQ
+that are documented in the
+[architecture overview](architecture-overview.html), we rely on the
+versions of those packages provided alongside the Linux distribution
+on which Zulip is deployed (for Debian and Ubuntu, this means `apt`).
+Since we don't control the versions of these dependencies, we avoid
+relying on specific versions of these packages wherever possible.
+
+The exact lists of `apt` packages needed by Zulip are maintained in
+two major places:
```

Comment:
This looks to me like two major places (puppet, and provision) and one minor one (VENV_DEPENDENCIES). I'd give separate bullets to each place, especially the two major ones. It's important to know that both exist.